### ISSUE TITLE
Implement accessibility pattern for help buttons/text

### ISSUE THREAD

**Issue by jmmcnj** [18F/dol-whd-14c#282]
As a [type of user] I would like help button/text to follow good accessibility patterns, as documented below: 

#### Functionality

@davoodharun Per our discussion, please add some Angular magic to get the help buttons and help text working and accessible. Details below. (- L. Sheu)

- We only need to make the help icons/text work on the **Application Page** for now. This work may break the help icons on other pages, but I will remediate as I update patterns on those pages.
- I have pushed my latest updates for the Application Info page to this branch: **lsheu-a11y-fixes-app-info**
- Relevant files: 
\DOL.WHD.Section14c.Web\src\modules\components\sectionAppInfo\sectionAppInfoTemplate.html
\DOL.WHD.Section14c.Web\src\modules\components\formSection\formSectionDirective.js
- These three areas of App Info have Help Buttons: "What Type of Application Is This?" (form question), "What type of establishment(s) are covered by this request..." (multiple help buttons in a repeater), and "Application Contact Person" (section heading)
- Please see the following punchlist and html examples below:

- [x] All Help Text blocks start hidden. When clicked, the Help Button should show/hide (toggle) its associated Help Text.  **Note** There is a css rule in forms.scss, line 169 which starts the .dol-help-text divs at display: none.
- [x] When clicked, the Help Button attribute **aria-expanded** should toggle between false/true. It should start at false. False is for when the Help Text is hidden, True for when the Help Text is shown.
- [x] The Help Button attribute **aria-controls** should have a value that matches the ID of the associated Help Text div. 
- [x] There is a link at the top right of the form titled **Show Help for All Items**. This link should toggle all the Help Text on the page open or closed.
- [x] The App Info form question **What type of establishment(s) are covered by this request for authority to employ workers with disabilities for?** is a repeater. Each checkbox label has a Help Button. The Help Button and Help Text need to work in the same way as other areas of the form. 

HTML example for the new Help Button:
`<button type="button" class="dol-help-link" aria-controls="[ID of help text div]" aria-expanded="[toggle false/true]">Help with this question</button>`

HTML example for the Help Text block:
`<div id="[ID]" class="dol-help-text">[help text goes here]</div>`

## Acceptance Criteria
Meets accessibility criteria as documented below

#### Relevant Accessibility Criteria

Keyboard Accessibility
[ ] 1.1 An interactive element or function can be accessed or activated by keyboard
[ ] 1.2 keyboard "trap" not found
[ ] 1.5 The visual focus can be determined at all times.
[ ] 1.9 the tab order is logical

Web: Forms- Web forms include controls (checkboxes, radio buttons etc.), and editable content (text input, select options etc.).
[ ] 2.1 All form fields and their instructions and cues have HTML (‘Label for’ and ‘ID’ are used or TITLE is descriptive) or ARIA for association.
[ ] 2.4 All ARIA form fields have a NAME property that contains all of the instructions.
[ ] 2.5 All ARIA form fields have correct Role, State, and Value properties.

Web: Links and User Controls
Links and/or user controls must have meaningful names that describe the unique destination, function, and/or purpose of the control for assistive technology.
[ ] 3.1 All links have a unique and meaningful description.
[ ] 3.2 All scripted elements have a unique and meaningful description.

**Comment by ltsheu** [comment]
@davoodharun The "Show Help for All Items" link needs the aria-expanded attribute. (Starting at false and toggling true/false as the help text blocks are turned on and off.)

**Comment by ltsheu** [comment]
@davoodharun As we discussed, Shannon and I decided that we do not need to add the aria-expanded attribute for "Show Help for All Items". We will be hiding the "Show Help for All Items" link from screen readers. I am moving this chore to Review/QA.
